### DISCUSSION TITLE
whitelist message

### DISCUSSION THREAD

**Discussion by ghost** [SkriptLang/Skript#3985] [Suggestions]
Skript should add something like { [the] whitelist( |-)message }
and it would edit spigot.yml or it would make a custom message

**Comment by abandonedaccount6235** [comment]
The best way to handle this would probably be creating your own whitelist functionality. I looked around a bit and it seems like Spigot doesn't provide any API to change the whitelist message besides editing that file.

**Reply by ghost** [reply]
> The best way to handle this would probably be creating your own whitelist functionality. I looked around a bit and it seems like Spigot doesn't provide any API to change the whitelist message besides editing that file.

creating own whitelist functionality is easy with 2 lines, but that wouldn't delete the player data and if someone has 
on join:
  some set variable thing
it would destroy it if they forgot to do it they would have to reset the variable if the variable was a list, if it was a nested variables they would have to go to variables.csv and delete it 1 by one if they have some variables they don't want to delete

**Reply by ghost** [reply]
> > The best way to handle this would probably be creating your own whitelist functionality. I looked around a bit and it seems like Spigot doesn't provide any API to change the whitelist message besides editing that file.
> 
> creating own whitelist functionality is easy with 2 lines, but that wouldn't delete the player data and if someone has
> on join:
> some set variable thing
> it would destroy it if they forgot to do it they would have to reset the variable if the variable was a list, if it was a nested variables they would have to go to variables.csv and delete it 1 by one if they have some variables they don't want to delete

so editing the file would be the best if you guys add it

**Reply by abandonedaccount6235** [reply]
unfortunately it's not really viable to edit the file, because even if we did do that it wouldn't take affect until the server restarts which isn't really what most people would want.

**Reply by TheDGOfficial** [reply]
Can be done with updating the value manually via reflection in SpigotConfig, don't know if it saves changes to file though most probably not so updating the file will still be needed for persistent changing.

As Skript requires Spigot already this not much of a problem except maintaining the reflection and file code. If spigot.yml or SpigotConfig format changes, our code will break. Not a reliable way as said above, so you can open a feature request to Spigot instead, but if anyone wants to implement this anyway, can be done with a reflection & snakeyaml logic.

**Reply by lilrosalyn** [reply]
Functionality doesn't seem super useful since it probably takes just as much or less time than manually updating the value in spigot.yml yourself

**Comment by TPGamesNL** [comment]
I thought this was already doable with [the kick event](https://skriptlang.github.io/Skript/events.html#kick) and [the kick message](https://skriptlang.github.io/Skript/expressions.html#ExprMessage), but apparently the kick message is not the same thing as the kick reason, see [PlayerKickEvent](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/player/PlayerKickEvent.html): it has `getReason`/`setReason` and `getLeaveMessage`/`setLeaveMessage`. This can be added as another enum value to `ExprMessage`: 
https://github.com/SkriptLang/Skript/blob/91a4e00f3e628fc0d1cbbaa0f5cb477e4b29134d/src/main/java/ch/njol/skript/expressions/ExprMessage.java#L106-L123
